src/jit/rationalize.cpp

```diff
@@ -1974,7 +1975,10 @@ Compiler::fgWalkResult Rationalizer::SimpleTransformHelper(GenTree **ppTree, Com
             break;
 
         case GT_LDOBJ:
-            RewriteLdObj(ppTree, data);
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+            if (varTypeIsSIMD(tree))
```

Comment:
I left it this way only because we have an assert that checks that we only have SIMD types here, so perhaps instead the if(varTypeIsSIMD(tree)) belongs in the RewriteLdObj method.